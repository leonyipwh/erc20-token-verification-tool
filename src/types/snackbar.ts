export enum ISnackbarType {
  ERROR = 'error',
  SUCCESS = 'success'
}

export interface ISnackbar {
  type: ISnackbarType;
  text?: string;
  open: boolean;
}

export interface ISnackbarContext {
  snackbarState: ISnackbar;
  setSnackbarState: (value: ISnackbar) => void;
}
