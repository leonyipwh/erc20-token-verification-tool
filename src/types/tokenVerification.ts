import { InfuraProvider } from '@ethersproject/providers';

export interface INetwork {
  name: string;
  subdomain: string;
}

export interface ITokenForm {
  network: INetwork;
  address: string;
  provider?: InfuraProvider;
}

export interface ITokenContext {
  submitForm: ITokenForm;
  setSubmitForm: (value: ITokenForm) => void;
}

export interface IProfileData {
  label: string;
  value: number | string | undefined;
}
