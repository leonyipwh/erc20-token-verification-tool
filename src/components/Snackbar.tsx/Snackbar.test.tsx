import { render, screen } from '@testing-library/react';
import Snackbar from './index';
import SnackbarContext from '../../context/snackbarContext';
import { ISnackbarType } from '../../types/snackbar';

describe('<Snackbar>', () => {
  it('should display error snackbar', () => {
    const snackbarState = {
      type: ISnackbarType.ERROR,
      open: true,
      text: 'this is error message'
    };
    const { asFragment } = render(
      <SnackbarContext.Provider value={{ snackbarState, setSnackbarState: () => {} }}>
        <Snackbar />
      </SnackbarContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display success snackbar', () => {
    const snackbarState = {
      type: ISnackbarType.SUCCESS,
      open: true,
      text: 'this is success message'
    };
    const { asFragment } = render(
      <SnackbarContext.Provider value={{ snackbarState, setSnackbarState: () => {} }}>
        <Snackbar />
      </SnackbarContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should hide snackbar', () => {
    const snackbarState = {
      type: ISnackbarType.ERROR,
      open: false,
      text: 'this is snack bar'
    };
    render(
      <SnackbarContext.Provider value={{ snackbarState, setSnackbarState: () => {} }}>
        <Snackbar />
      </SnackbarContext.Provider>
    );
    const snackbar = screen.queryByText(snackbarState.text);
    expect(snackbar).not.toBeInTheDocument();
  });
});
