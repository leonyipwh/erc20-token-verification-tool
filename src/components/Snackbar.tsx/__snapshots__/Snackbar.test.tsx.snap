// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar> should display error snackbar 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight css-1arijsr-MuiSnackbar-root"
    role="presentation"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-3yjvh5-MuiPaper-root-MuiAlert-root"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-acap47-MuiAlert-message"
      >
        this is error message
      </div>
      <div
        class="MuiAlert-action css-ki1hdl-MuiAlert-action"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          title="Close"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Snackbar> should display success snackbar 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight css-1arijsr-MuiSnackbar-root"
    role="presentation"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-sp7f89-MuiPaper-root-MuiAlert-root"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="SuccessOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-acap47-MuiAlert-message"
      >
        this is success message
      </div>
      <div
        class="MuiAlert-action css-ki1hdl-MuiAlert-action"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          title="Close"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
